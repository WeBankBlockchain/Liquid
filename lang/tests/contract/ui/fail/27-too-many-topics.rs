use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(event)]
    struct TestEvent {
        i: i8,
        #[liquid(indexed)]
        b: bool,
        x: i16,
        #[liquid(indexed)]
        y: i32,
        #[liquid(indexed)]
        z: i64,
        #[liquid(indexed)]
        s: String,
    }

    #[liquid(methods)]
    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) -> () {
            self.env().emit(TestEvent {
                i: 0,
                b: true,
                x: 0,
                y: 0,
                z: 0,
                s: String::from("123"),
            });
        }
    }
}

fn main() {}
