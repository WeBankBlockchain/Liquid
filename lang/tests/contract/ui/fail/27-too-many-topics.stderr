error: the number of topics should not exceed 3 in `liquid(event)` struct
  --> $DIR/27-too-many-topics.rs:18:9
   |
18 | /         #[liquid(indexed)]
19 | |         s: String,
   | |_________________^
